redis/src/main/java/org/togglz/redis/RedisStateRepository.java

```diff
@@ -0,0 +1,171 @@
+package org.togglz.redis;
+
+import org.togglz.core.Feature;
+import org.togglz.core.repository.FeatureState;
+import org.togglz.core.repository.StateRepository;
+import org.togglz.core.repository.util.DefaultMapSerializer;
+import org.togglz.core.repository.util.MapSerializer;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.JedisPool;
+import redis.clients.jedis.JedisPoolConfig;
+import redis.clients.jedis.Protocol;
+
+import java.util.Map;
+
+/**
+ * A state repository which stores the feature state in Redis.
+ * <p>
+ * The class provides a builder which can be used to configure the state repository instance
+ * (e.g. Redis hostname, Redis/Jedis configuration, serialization).
+ * </p>
+ *
+ * @author Cosmin Rentea
+ */
+public class RedisStateRepository implements StateRepository {
+
+    public static final String KEY_PREFIX = "togglz-";
+    public static final String ENABLED_FIELD = "enabled";
+    public static final String STRATEGY_FIELD = "strategy";
+    public static final String PARAMETERS_FIELD = "parameters";
+
+    protected final JedisPool jedisPool;
+    protected final JedisPoolConfig jedisPoolConfig;
+    protected final String hostname;
+    protected final String keyPrefix;
+    protected final MapSerializer mapSerializer;
+
+    public RedisStateRepository() {
```

Comment:
Just wondering. Why do you provide this default constructor? Is this useful for users? Why not force them to configure the repository correctly using the builder?